.github/workflows/ci.yml

```diff
@@ -20,20 +20,18 @@ jobs:
     # begin macro
 
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

Comment:
> Pinned all actions to avoid certain classes of vulnerabilities.

What vulnerabilities are you expecting to prevent?

Reply:
the ability of the action authors to publish an update with malicious code running with full access to the CI and ability to silently undermine correctness and security of what's being built.

Reply:
if Github makes an update to `actions/checkout` that is malicious, I expect they would halt all the runners before it affected us. The looks like noise to me.

Reply:
> * Pinned all actions to avoid certain classes of vulnerabilities.

Let's only pin 3rd party actions. GH maintained ones should be out of scope for this class of vulnerability

Reply:
I think that kind of misses the point...

FWIW, GitHub itself recommends you pin SHAs.

Reply:
> GitHub is a third party

I don't think that page GitHub is referring to itself as a third party. Even if it were, what you cite also says,
>Although pinning to a commit SHA is the most secure option, specifying a tag is more convenient and is widely used. If you’d like to specify a tag, then be sure that you trust the action's creators. The ‘Verified creator’ badge on GitHub Marketplace is a useful signal, as it indicates that the action was written by a team whose identity has been verified by GitHub. Note that there is risk to this approach even if you trust the author, because a tag can be moved or deleted if a bad actor gains access to the repository storing the action.

`v4` is a tag and thus the status quo satisfies the advice you cite.

Reply:
@turadg 

> Note that there is risk to this approach even if you trust the author, because a tag can be moved or deleted if a bad actor gains access to the repository storing the action.

Reply:
I don't worry GitHub will become a bad actor, but "third party" is strictly a relative term. There's no way GitHub is _not_ a third party to me or you _unless_ you work for GitHub.

Reply:
yeah I read the text I quoted. I'm done with this debate.

Reply:
This is about a GH workflow, running on GH infra. By definition endo is trusting that infra, and whatever latest deployments there may be of it. When I said 3rd party, I meant to include GH as a 1st party since that's where the root is.

While I agree the possibility of a branch in a `actions` org repo being updated maliciously is higher than the actions workflow infra itself being updated maliciously, I still consider both extremely unlikely, and not worth defensing against.

I will not die on that hill however.

Reply:
I am dying on this hill. 😄

Reply:
(we have some [nascent tooling](https://github.com/LavaMoat/LavaMoat/tree/main/packages/git-safe-dependencies) which checks this sort of thing as well, but it needs docs explaining why it does what it does. @naugtur 😉)

Reply:
As long as I'm not the one maintaining this, I'm ok with shas here.

Reply:
RE maintaining, do we plan on introducing Renovate to endo?

since i don't think dependabot can PR yml SHA updates (i could be wrong)

Reply:
IMO, we should, but others may disagree. I'll throw something on the meeting agenda.